libraries/AP_HAL/utility/RingBuffer.cpp

```diff
@@ -6,9 +6,14 @@
   implement a simple ringbuffer of bytes
  */
 
+static inline uint32_t align_power2(uint32_t u)
+{
+    return 1U << ((sizeof(u) * 8) - __builtin_clz(u - 1));
+}
+
 ByteBuffer::ByteBuffer(uint32_t _size)
 {
-    size = _size;
+    size = align_power2(_size);
```

Comment:
aligning to a power of 2 is a bad idea. First off, on platforms with small amounts of memory a buffer of 5k might be ideal, and 8k can be too much memory. Secondly, ByteBuffer is also used by ObjectBuffer which creates a buffer of a specific size and expects it to be that size.
Long ago the macros which preceded this code assumed power of 2 size. I changed it to not assume power of 2 as we needed buffers of flexible sizes. The CPU cost of the modulus is tiny.